### ISSUE TITLE
[Q] Visualize new changes in the logged metrics after resuming a run

### ISSUE THREAD

**Issue by EmanueleFrancazi** [wandb/wandb#3373]
Hi,
I use pytorch and I'm currently trying to add the possibility to resume a finished simulation; the code seem to work fine [load and save a model](https://pytorch.org/tutorials/beginner/saving_loading_models.html).
Since I also use wandb I'd like to retrieve also all the logs started on the  first half of the simulation.
Following one of the possibilities suggested [here](https://docs.wandb.ai/guides/track/advanced/resuming) I modified the initialization as follows:

```
if (StartMode=='BEGIN'):
    StartEpoch = 0
    StartIterationCounter=0
    ProjId = wandb.util.generate_id()
elif(StartMode=='RECAP'):
    #here we retrieve the model and state of the old checkpoint; first initialize the model and optimizer, then load the dictionary locally.
    checkpoint = torch.load(FolderPath +'/model.pt') #load the model (substitute with the right pathas argument)
    NetInstance.model.load_state_dict(checkpoint['model_state_dict'])
    NetInstance.optimizer.load_state_dict(checkpoint['optimizer_state_dict'])
    ProjId = checkpoint['proj_id']
    StartEpoch = checkpoint['epoch']
    StartIterationCounter = checkpoint['step']   
    n_epochs = n_epochs + StartEpoch
    #we have also to recall the old variables:
        #- we load them in different container; in this way to fill the new files' variable we can directly charge the old ones and the new one
    NetInstance.RecallOldVariables()

run = wandb.init(project= ProjName,
           group =  GroupName,#with the group parameter you can organize runs divide them into groups
           notes="some comments",
           entity= "user_name"
           name = RunName,
           id = ProjId, #you can use id to resume the corresponding run; to do so you need also to express the resume argument properly
           resume="allow"
           #,reinit=True #If you're trying to start multiple runs from one script, add two things to your code: (1) run = wandb.init(reinit=True): Use this setting to allow reinitializing runs
           )

NetInstance.CustomizedX_Axis() #set the customized x-axis in a automatic way for all the exported charts

wandb.config = {
  "Algorithm": Dynamic,
  "Imbalance_ratio":UnbalanceFactor,
  "learning_rate": learning_rate,
  "epochs": n_epochs,
  "batch_size": batch_size,
  "Number_of_channel_grous_(group_norm)": group_factor,
  "Percentage_of_Elements_Dropped_out" : dropout_p,
  "Imbalance_ratio":UnbalanceFactor,
  "architecture": args.Architecture,
  "Dataset":args.Dataset
}
```
where `StartMode` is the flag that establish if we are starting a run from 0 or resuming an old one; in the first case we create a new unique id with `wandb.util.generate_id()`, while in the second case we use the one saved from the first half of the run.
`NetInstance` is just the instance of a class from where my model is defined.
Proceding in this way I can resume the previous run and update for example the file uploaded on wandb.

On the other hand also if in the new simulation (resuming the old one, i.e. with `StartMode=='RECAP'`) I try to add new values to the former logged metrics using `wandb.log('old metrics': new values)` they don't appear on the dashboard's chart.
What am I missing?

[EDIT]
Form this [link](https://docs.wandb.ai/guides/track/advanced/resuming) I think that wandb.restore is the step that misses in my code:
> This will allow you to log new historical values for your metrics to a run starting from where you left off 

But when I click on the [link](https://docs.wandb.ai/guides/track/advanced/save-restore#examples-of-wandb.restore) to get more info, the page only explain how to restore file, it doesn't say anything about the metrics

**Comment by nate-wandb** [comment]
Hi @EmanueleFrancazi, is the `ProjId` variable that you get from the Torch checkpoint a Weights and Biases `run_id`?

Your code seems to be correct, so I'm wondering if this is getting logged but not to the correct run. If you use `resume="allow"` it will create a new run if that run_id doesn't exist. If you click the link provided in the terminal to your run does it take you to the run you were trying to resume?

As far as how `wandb.restore()` works, that link may be a bit misleading. The purpose of the function is to restore files such as the model weights from W&B. It looks like in your case that you are storing your model locally on your computer so I'm not sure you would need that here. [Here ](https://docs.wandb.ai/guides/track/advanced/save-restore#restoring-files)is a better example of the usage of `wandb.restore()`

Lastly here is some minimal working code that correctly appends new metric data to one my runs with the `run_id` "o0wws5p1"

```import wandb
run = wandb.init(project='testing', name= "worthy-armadillo-3", resume='allow', id = 'o0wws5p1', reinit=True)
run.log({'metric': 1})
run.finish()
```

It seems that you have all of the elements needed in your code which is why I was wondering if there was an issue with the `run_id` you are passing in

Thank you,
Nate

**Comment by EmanueleFrancazi** [comment]
>  is the ProjId variable that you get from the Torch checkpoint a Weights and Biases run_id?

Hi nate,
yes at the end of the simulation I have:
```
torch.save({'model_state_dict' : NetInstance.model.state_dict(),
            'optimizer_state_dict' : NetInstance.optimizer.state_dict(), #When saving a general checkpoint, you must save more than just the model’s state_dict. It is important to also save the optimizer’s state_dict, as this contains buffers and parameters that are updated as the model trains.
            'epoch': n_epochs,  #Other items that you may want to save are the epoch 
            'step': IterationCounter,  #or number of iteration
            'proj_id': ProjId,
            }, FolderPath +'/model.pt')``
```
you can see that one of the saved object is the ProjId. I also checked it by printing and comparing the `id` of the starting run and of its resuming; I got the same name. This is also confirmed by the fact that the files containing measures (which I save, at the end of each simulation, on wandb with the command `wandb.save(FolderPath+'/*.txt')`) are correctly updated after the resuming of the run has completed.

**Comment by nate-wandb** [comment]
@EmanueleFrancazi I wanted to follow up and see if you were still experiencing this issue?

**Comment by v-iashin** [comment]
hey, @nate-wandb , I think I've got a similar issue.

After initing the `wandb` with `resume='must'` and `id`, the UI does not update the charts with the values obtained during the resumed run and they stay the same as they were except for the `System` charts where they duplicate (because x is the time relative to start of the run).

Anyway, 
* `Summary` in `Overview` is updated
* `wandb-summary.json` (in the Files tab) is not updated
* `wandb-summary.json` (locally) is updated

Here is one of the charts (note x = 6700 it is where it failed before resuming)
![Screenshot 2023-06-22 at 14 45 41](https://github.com/wandb/wandb/assets/14952597/1c079c23-f3ff-4204-8dbf-bc5a28db960e)

The updated `Summary` in `Overview`
![Screenshot 2023-06-22 at 14 48 36](https://github.com/wandb/wandb/assets/14952597/70630e6a-8321-4943-b858-ef7249246132)
